docs/runtimes/node-affinity.md

```diff
@@ -1,96 +1,347 @@
-MLRun jobs which are stateful, and are assigned to run on spot nodes, may be subject to interruption 
-and will to be designed so that the job/function state will be saved when scaling to zero.
+- V3IO containers: when running on the Iguazio system, pods have access to the underlying V3IO shared storage. This option mounts a V3IO container or a subpath within it to the pod through the V3IO FUSE driver.
+- PVC: Mount a Kubernetes persistent volume claim (PVC) to the pod. The persistent volume and the claim need to be configured beforehand.
+- Config Map: Mount a Kubernetes Config Map as local files to the pod.
+- Secret: Mount a Kubernetes secret as local files to the pod.
 
-## Node selector
-Using the **Node Selector** you can assign MLRun jobs to specific nodes within the cluster. 
-**Node Selector** is available for all modes of deployment in the platform including the platform UI, 
-command line, and programmable interfaces.
+For each of the options, a name needs to be assigned to the volume, as well as a local path to mount the volume at (using a Kubernetes Volume Mount). Depending on the type of the volume, other configuration options may be needed, such as an access-key needed for V3IO volume.
+
+See more about [Kubernetes Volumes](https://kubernetes.io/docs/concepts/storage/volumes/).
+
+MLRun supports the concept of volume auto-mount which automatically mounts the most commonly used type of volume to all pods, unless disabled. See more about [MLRun auto mount](function-storage).
+
+### UI configuration
+
+You can configure Volumes when creating a job, rerunning an existing job, and creating an ML function.
+Modify the Volumes for an ML function by pressing **ML functions**, then **<img src="../_static/images/kebab-menu.png" width="25"/>** 
+of the function, **Edit** | **Resources** | **Volumes** drop-down list. 
+
+Select the volume mount type: either Auto (using auto-mount), Manual or None. If selecting Manual, fill in the details in the volumes list 
+for each volume to mount to the pod. Multiple volumes can be configured for a single pod.
+
+### SDK configuration
+
+Configure volumes attached to a function by using the `apply` function modifier on the function. 
+
+For example, using v3io storage:
+```
+# import the training function from the marketplace (hub://)
+train = mlrun.import_function('hub://sklearn_classifier')# Import the function:
+open_archive_function = mlrun.import_function("hub://open_archive")
+
+# use mount_v3io() for iguazio volumes
+open_archive_function.apply(mount_v3io())
+```
+You can specify the v3io path to use for the volume (`remote`), the volume mount path (`volume_mounts`), and security credentials, for example: `mlrun.platforms.mount_v3io(name='v3io', remote='', mount_path='', access_key='', user='', secret=None, volume_mounts=None)`. See full details in [mount_v3io](../api/mlrun.platforms.html#mlrun.platforms.mount_v3io).
```

Comment:
You can specify a list of the v3io path to use and how they map inside the container ( using `volume_mounts`)... for example: `mlrun.mount_v3io(name='data', access_key='XYZ123..', volume_mounts=[mlrun.VolumeMount("/data", "projects/proj1/data")])`